### ISSUE TITLE
Error on startup

### ISSUE THREAD

**Issue by QuillMcGee** [GoodAI/BrainSimulator#4]
When I attempt to start Brain simulator, an error message appears saying "An error occured while initializing simulation. Please make sure you have a supported CUDA-enabled graphics card and appropriate drivers. Technical details: InitializationFailed: Initialization of CUDA failed." I have successfully been using CUDA on this computer (windows 7, graphics card GT 420m) for some time (mostly Theano and other gpu-accelerated python stuff). Is there anything I can do to fix this, or any diagnostic information I could provide to help you help me/help you correct whatever bug is causing this?

**Comment by deranjer** [comment]
Exact same issue here, but I have an ATI card, so I may just be out of luck?

**Comment by QuillMcGee** [comment]
Yeah, only NVIDIA cards support CUDA, I'm afraid.

**Comment by PremekPaska** [comment]
Hi Quill, we are sorry to hear that it's not working for you.

First, I'd like to make sure that your Windows is 64-bit, *Brain Simulator only supports 64-bit systems*.

If it is 64-bit, than I'd suggest that you install CUDA7 SDK that you can download here: https://developer.nvidia.com/cuda-downloads

You also need to delete (or better just move) the CUDA libraries that are installed with Brain Simulator, so that the system-wide libraries are used. (The default location is `C:\Program Files\GoodAI\Brain Simulator`.) There are 4 libraries with names starting with *lowercase* "cu", specifically `cublas64_70.dll`, `cudart64_70.dll`, `cufft64_70.dll` and `curand64_70.dll`. (Uppercase Cu* libraries are ManagedCuda libraries and need to stay there.)

Please try to run Brain Simulator with this setup and let us know if it worked. Thanks!

**Comment by PremekPaska** [comment]
Also, if you have both integrated *and* standalone (hi-end) graphics, **make sure that the standalone graphics is enabled for Brain Simulator**. Thanks for `eobando` who pointed this out at the [forum](http://forum.goodai.com/index.php?threads/bug-report-initialization-error.8/#post-23).

**Comment by QuillMcGee** [comment]
I do have both integrated and standalone graphics. All the other GPU-computing I've done has used the standalone automatically, how can I ensure that Brain Simulator will do so as well?

**Comment by PremekPaska** [comment]
The user that pointed this out described is as follows: *In my card settings, I had to add Brain Simulator into the list of programs to use the high-end card. Once done, all is GoodAI :)*

Also, for another user, installing the CUDA SDK solved the issue. Have you tried that?

**Comment by QuillMcGee** [comment]
The CUDA 7 SDK did not install successfully even after several attempts, and I forgot to post back here, however I have now successfully installed CUDA 7.5, at which point it threw a different error saying that curand64_70 could not be found, at which point I went to C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.5\bin, looked up the list of four dlls that brain simulator needs, and copied them and renamed the copies whatver64_70 rather than whatever64_75, and it booted properly. 

This issue has been resolved.

**Comment by PremekPaska** [comment]
Thanks for your report. It's interesting that it works at all :)

The question is why CUDA 7 failed to install and 7.5 (which is now in RC) works...

**Comment by QuillMcGee** [comment]
Oh, and I tried to install 7 again after installing 7.5 in case it was a problem with the existing 6.5 installation, no dice. 

Also, "works" might be a bit strong. I did download and unpack 7 and copy over the four dlls that Brain Simulator needs to 7.5's bin after encountering problems, without changing anything, so at least that probably wasn't the issue. I'll be submitting another ticket soon, it was just a different problem so I closed this one.
